pkg/frontend/read_path/query_frontend/query_frontend.go

```diff
@@ -148,3 +187,75 @@ func (q *QueryFrontend) QueryMetadata(
 
 	return md.Blocks, nil
 }
+
+// hasUnsymbolizedProfiles checks if a block has unsymbolized profiles
+func (q *QueryFrontend) hasUnsymbolizedProfiles(block *metastorev1.BlockMeta) bool {
+	matcher, err := labels.NewMatcher(labels.MatchEqual, metadata.LabelNameUnsymbolized, "true")
+	if err != nil {
+		return false
+	}
+
+	return len(slices.Collect(metadata.FindDatasets(block, matcher))) > 0
```

Comment:
@aleks-p Can you help me verify my assumption here please.
1. Blocks contain multiple datasets and so a block may have both unsymbolized and symbolized datasets.
2. I believe the `MetadataQueryService.QueryMetadata` does not return all the datasets present in the block ,but only the ones that matched query.  I've made this conclusion from reading this code
https://github.com/grafana/pyroscope/blob/d564c76454a93f75d008c3a0b223102cdaf55c5c/pkg/experiment/metastore/index/query.go#L163

If my understanding is correct, than this  `hasUnsymbolizedProfiles` is fine. If the second assumption is not correct, then this check would degrade the symbolized query. 

I think everything is good here, just want another :eyes: to doublecheck.